storage/src/nodestore.rs

```diff
@@ -1569,3 +1638,115 @@ mod tests {
         println!("{:?}", immutable); // should not be reached, but need to consume immutable to avoid optimization removal
     }
 }
+
+#[cfg(test)]
+mod test_node_store_checker {
+    use super::*;
+    use crate::linear::filebacked::FileBacked;
+    use crate::{BranchNode, LeafNode};
+
+    use std::num::NonZeroUsize;
+    use tempfile::NamedTempFile;
+
+    const NODE_CACHE_SIZE: NonZeroUsize = NonZeroUsize::new(1000).unwrap();
+    const FREE_LIST_CACHE_SIZE: NonZeroUsize = NonZeroUsize::new(1000).unwrap();
+
+    // Helper function to wrap the node in a StoredArea and write it to the given offset. Returns the area size on success.
+    fn write_new_node(
+        file_backed: &FileBacked,
+        node: &Node,
+        offset: u64,
+    ) -> Result<usize, FileIoError> {
+        let node_size = NodeStore::<Arc<ImmutableProposal>, FileBacked>::stored_len(node);
+        let (area_index, node_area) = AREA_SIZES
+            .iter()
+            .enumerate()
+            .find(|(_, size)| **size >= node_size)
+            .unwrap();
+        let mut stored_area_bytes = Vec::with_capacity(*node_area as usize);
+        node.as_bytes(area_index as u8, &mut stored_area_bytes);
+        file_backed.write(offset, &stored_area_bytes)?;
```

Comment:
This is a lot of work, and allocates space that will never get used. probably easier to leverage area_size_to_index.

I'd say the algorithm should be:
 - Call node.as_bytes() to get the length
 - Call filebacked_write(offset, the_bytes)
 - Call area_size_to_index(the_bytes.len())
 - Return AREA_SIZES[whatever_area_size_to_index returned]
 
This will make you immune to changes to the way allocations work in the future.

Reply:
This code preallocates some AREA_SIZE value for `stored_area_size`, but you'll only ever write `node_size` bytes to `stored_area_size`. If you were to fix this code, you should only allocate `node_size` instead of `node_area` on line 1666, but as mentioned you shouldn't fix this code, you should instead use the algorithm I described above anyway.